src/index.ts

```diff
@@ -330,7 +364,16 @@ export class CruxClient extends CruxPayPeer {
 
             // Generating the identityClaim
             if (this._payIDClaim) { await (this._payIDClaim as PayIDClaim).decrypt(); }
-            const identityClaim = this._payIDClaim ? {secrets: this._payIDClaim.identitySecrets} : await (this._nameService as nameService.NameService).generateIdentity();
+
+            let identityClaim: nameService.IIdentityClaim;
+            if (this._payIDClaim) {
+                identityClaim = {secrets: this._payIDClaim.identitySecrets};
+            } else if (this._keyPair) {
+                identityClaim = {secrets: {identityKeyPair: this._keyPair}};
+            } else {
+                identityClaim = await (this._nameService as nameService.NameService).generateIdentity();
```

Comment:
Is it safe to assume that identityClaim format is the same in all cases here?

Reply:
Yes, format is same.